libraries/botframework-schema/src/sharepoint/cardView/cardViewParameters.ts

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { CardImage } from './cardImage';
+import { CardBarComponent } from './cardBarComponent';
+import { CardButtonComponent } from './cardButtonComponent';
+import { CardSearchBoxComponent } from './cardSearchBoxComponent';
+import { CardSearchFooterComponent } from './cardSearchFooterComponent';
+import { CardTextComponent } from './cardTextComponent';
+import { CardTextInputComponent } from './cardTextInputComponent';
+
+/**
+ * The actions-only footer parameters for the card view.
+ */
+export type CardViewActionsFooterParameters =
+    | [CardButtonComponent]
+    | [CardButtonComponent, CardButtonComponent]
+    | undefined;
+
+/**
+ * The footer parameters for the card view. Can contain either 0 to 2 buttons or a single text input.
+ */
+export type CardViewFooterParameters = CardViewActionsFooterParameters | [CardTextInputComponent];
+
+/**
+ * Base parameters for the card view.
+ */
+export interface BaseCardViewParameters {
+    /**
+     * Card view title area (card bar) components.
+     */
+    cardBar: [CardBarComponent];
+}
+
+/**
+ * The parameters for the card view with text or empty body.
+ */
+export interface TextCardViewParameters extends BaseCardViewParameters {
```

Comment:
Can any of these types be reused from an already published npm packge?

Reply:
I don't want to have a reference to our packages for 2 reasons:
1. we will need to tweak the imported types. For example, remove some fields like `_events` and also remove actionable elements like event handlers.
2. We will tie to the ACE core release cadence. Bots (I mean Bot ACEs) have its own candence. Although it's mostly a copy-paste, it seems reasonable here.